python/lsst/daf/ingest/ingestCatalogTask.py

```diff
@@ -0,0 +1,580 @@
+    if sz > 65535:
+        # If necessary, longer strings could be ingested as TEXT.
+        raise RuntimeError("String field is too large for ingestion")
+    elif sz == 0:
+        raise RuntimeError("String field has zero size")
+    # A string containing trailing spaces cannot round-trip to a CHAR
+    # column and back. Therefore, use VARCHAR. Also, assume strings are
+    # ASCII for now.
+    return "VARCHAR({}) CHARACTER SET ascii COLLATE ascii_bin NOT NULL".format(sz)
+
+def _sqlTypeForArray(formatChar, field):
+    """Helper function for computing the SQL column type of an array valued field.
+    See https://docs.python.org/2/library/struct.html#format-characters for possible
+    format characters. The formatChar passed in must correspond to the array element
+    type.
+    """
+    sz = field.getSize()
+    if sz == 0:
+        return "BLOB NOT NULL"
+    sz *= struct.calcsize("<" + formatChar)
+    if sz > 65535:
+        raise RuntimeError("Array field is too large for ingestion")
+    return "BINARY({}) NOT NULL".format(sz)
+
+
+"""A dictionary mapping afw field type strings to corresponding formatters.
+
+This dictionary is used by IngestCatalogTask to determine how to format each
+type of field in the input catalog.  If new field types are added to the afw
+table library, they should also be added here.
+"""
+fieldFormatters = dict(
+    U = FieldFormatter(lambda f: "SMALLINT UNSIGNED NOT NULL", lambda v: str(v)),
+    I = FieldFormatter(lambda f: "INT NOT NULL", lambda v: str(v)),
+    L = FieldFormatter(lambda f: "BIGINT NOT NULL", lambda v: str(v)),
+    F = FieldFormatter(lambda f: "FLOAT", lambda v: _formatNumber("{:.9g}", v)),
+    D = FieldFormatter(lambda f: "DOUBLE", lambda v: _formatNumber("{:.17g}", v)),
+    Flag = FieldFormatter(lambda f: "BIT NOT NULL", lambda v: "1" if v else "0"),
+    Angle = FieldFormatter(lambda f: "DOUBLE", lambda v: _formatNumber("{:.17g}", v.asDegrees())),
+    String = FieldFormatter(_sqlTypeForString, _formatString),
+    ArrayU = FieldFormatter(lambda f: _sqlTypeForArray("H", f), lambda v: _formatArray("H", v)),
+    ArrayI = FieldFormatter(lambda f: _sqlTypeForArray("i", f), lambda v: _formatArray("i", v)),
+    ArrayF = FieldFormatter(lambda f: _sqlTypeForArray("f", f), lambda v: _formatArray("f", v)),
+    ArrayD = FieldFormatter(lambda f: _sqlTypeForArray("d", f), lambda v: _formatArray("d", v)),
+)
+
+
+def canonicalizeFieldName(fieldName):
+    """Return a MySQL-compatible version of the schema field name, for now by
+    changing any non-word characters to underscores.
+    """
+    return re.sub(r"[^\w]", "_", fieldName)
+
+def quoteIdentifier(identifier):
+    """Return a MySQL compatible version of the given identifier by quoting
+    it with back-ticks (and escaping embedded back-ticks).
+    """
+    return "`" + identifier.replace("`", "``") + "`"
+
+def aliasesFor(name, mappings):
+    """Compute and return the set of possible aliases for the given field name.
```

Comment:
I thought I saw that aliases were implemented as an aid to schema migration rather than something that we are going to want to be an integral part of the system. I do worry that a system that encourages alias usage could lead to surprise and confusion downstream. We want code to be using a standardized set of column names and use aliases to enable us to map non-standard tables to the standard. Or I've misunderstood completely.